airgun/views/host.py

```diff
@@ -0,0 +1,318 @@
+    SatTable,
+)
+
+
+class TableActions(View):
+    """Interface table has Action column that contains only two buttons,
+    without any extra controls, so we cannot re-use any existing widgets
+    """
+    edit = Text(".//button[contains(., 'Edit')]")
+    delete = Text(".//button[contains(., 'Delete')]")
+
+
+class HostInterface(View):
+    ROOT = ".//div[@id='interfaceModal']"
+    title = Text(".//h4[contains(., 'Interface')]")
+    submit = Text(".//button[contains(@onclick, 'save_interface_modal')]")
+    interface_type = FilteredDropdown(id='_type')
+    mac = TextInput(locator=".//input[contains(@id, '_mac')]")
+    device_identifier = TextInput(
+        locator=".//input[contains(@id, '_identifier')]")
+    dns = TextInput(locator=".//input[contains(@id, '_name')]")
+    domain = FilteredDropdown(id='_domain_id')
+    subnet = FilteredDropdown(id='_subnet_id')
+    subnet_v6 = FilteredDropdown(id='_subnet6_id')
+    ip = TextInput(locator=".//input[contains(@id, '_ip')]")
+    ipv6 = TextInput(locator=".//input[contains(@id, '_ip6')]")
+    managed = Checkbox(locator=".//input[contains(@id, '_managed')]")
+    primary = CheckboxWithAlert(locator=".//input[contains(@id, '_primary')]")
+    provision = CheckboxWithAlert(
+        locator=".//input[contains(@id, '_provision')]")
+    remote_execution = Checkbox(
+        locator=".//input[contains(@id, '_execution')]")
+    # when interface type is selected, some additional controls will appear on
+    # the page
+    interface_additional_data = ConditionalSwitchableView(
+        reference='interface_type')
+
+    @interface_additional_data.register('Interface')
+    class InterfaceForm(View):
+        virtual_nic = Checkbox(
+            locator=".//input[contains(@id, '_virtual')]")
+        virtual_attributes = ConditionalSwitchableView(
+            reference='virtual_nic')
+
+        @virtual_attributes.register(True, default=True)
+        class VirtualAttributesForm(View):
+            tag = TextInput(locator=".//input[contains(@id, '_tag')]")
+            attached_to = TextInput(
+                locator=".//input[contains(@id, '_attached_to')]")
+
+    @interface_additional_data.register('BMC')
+    class BMCForm(View):
+        username = TextInput(locator=".//input[contains(@id, '_username')]")
+        password = TextInput(locator=".//input[contains(@id, '_password')]")
+        provider = FilteredDropdown(id='_provider')
+
+    @interface_additional_data.register('Bond')
+    class BondForm(View):
+        mode = FilteredDropdown(id='_mode')
+        attached_devices = TextInput(
+            locator=".//input[contains(@id, '_attached_devices')]")
+        bond_options = TextInput(
+            locator=".//input[contains(@id, '_bond_options')]")
+
+    @interface_additional_data.register('Bridge')
+    class BridgeForm(View):
+        attached_devices = TextInput(
+            locator=".//input[contains(@id, '_attached_devices')]")
+
+    def after_fill(self, was_change):
```

Comment:
Why do you use `after_fill` instead of clicking submit inside entity helper? Seems like deviation from our style.

Reply:
because that submit action happens in the middle of `view.fill()` operation, so it is impossible to access that place from entity helper

Reply:
Sure. But not for `HostsChangeGroup` and `HostsChangeEnvironment `, please address there.